backend/src/main/java/org/cryptomator/hub/entities/GrantVaultAccessEvent.java

```diff
@@ -0,0 +1,53 @@
+package org.cryptomator.hub.entities;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.DiscriminatorValue;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.UUID;
+
+@Entity
+@Table(name = "grant_vault_access_event")
+@DiscriminatorValue(GrantVaultAccessEvent.TYPE)
+public class GrantVaultAccessEvent extends AuditEvent {
+
+	public static final String TYPE = "GRANT_VAULT_ACCESS";
+
+	@Column(name = "user_id")
+	public String userId;
+
+	@Column(name = "vault_id")
+	public UUID vaultId;
+
+	@Column(name = "authority_id")
+	public String authorityId;
```

Comment:
Regarding `userId` and `authorityId` it is unclear who granted access and whom was granted access.

Maybe we can find better names.

Reply:
Yeah, I just used the same naming pattern from other events. In this case, it's similar to `UpdateVaultMembershipEvent`. And also from all the other events because they all have a `user_id` to identify the user that caused this event. This could indeed be improved.

But should we leave `user_id` as it is? I don't think that the name is _that_ bad. Or should we refactor it to something new like `actor_id` or `created_by` (which doesn't really fit with the other columns like `timestamp`)? And I wonder if this column is a common one for _all_ events, then we could move it to `AuditEvent`.

Regarding `authority_id`: How about `granted_authority_id` for grant access? And for update vault membership, `updated_authority_id`?

Reply:
>I don't think that the name is that bad

I second that. If you give it one though, you notice, that only an actual user can create an event, hence `user_id` is always the cause while authority_id must be part of the target.

Reply:
> And I wonder if this column is a common one for _all_ events, then we could move it to `AuditEvent`.

One reason against this is that if we add "system" events in the future that are not caused by a logged-in user.

Reply:
Let's change it, while we still have the chance.

I like `actor_id` and then naming the columns of the target depending on the event. Another option would be to follow the JWT standard with `issuer`( == user_id) and `subject` (==authority_id).

Reply:
Fixed in 01b3888d7c19313a85bae37f4a1fb06421682198 by renaming `user_id` to be more specific, e.g., `granted_by` or `created_by`. I've left `authority_id` as is because it felt clear to me that this was the subject (as in other events that a vault or device is the subject).